pkg/constants/constants.go

```diff
@@ -121,6 +121,13 @@ const (
 	InferenceServiceDefaultLoggerPort = "8081"
 )
 
+// Labels to put on kservice
+const (
+	KServiceComponentLabel = "component"
```

Comment:
Do Component and Predictor signal the name information? Can we dedup?

Reply:
Component is one of predictor/transformer/explainer

Predictor is one of default/canary

Basically predictor would just signal whether the component is for a canary. So pretty much a flag but in the future there could be further options beyond canary if a/b tests or shadows were added

Reply:
Check the terminology here:
https://github.com/kubeflow/kfserving/tree/master/docs#concepts

I think we should use "Endpoint" and "Component"

Reply:
Have updated now